Cardinal Station (old Bunched station)

BarteG44 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request
Card Station is a station based around 4 central corridors running in cardinal directions. Despite being about as large as Fland station in footprint, it's departments are placed and laid out is such a way as to ensure player interaction.

Design principles:
-High traffic areas such as security, bridge, and medbay are all located at the ends of the corridors, forcing traffic to go trough the service-oriented center.
-Most rooms directly connected to the corridors also have plentiful windows facing them with optional shutters or curtains. This is done to allow anyone from the corridor to see and talk directly to people who may be busy in their workplace, who would usually be tucked away somewhere far from sight. The shutters are there to permit some privacy to the people working, while their prevalence will make it less likely for anyone to suspect them of shady business if they do choose to close their department off. It's not suspicious if everyone's doing it.
-Greenery is present in most places. Just like in real life, having some plants around makes the whole space more pleasant to look at.
-More gimmick areas such as an actual Theater, Barbershop, decently sized office and a maintenance night club
-to allow for more or less discrete travel between parts of maintenance, there are 6 rooms linked in pairs with dispsal pipes
## Why It's Good For The Game

Box, Meta, and delta are pretty much just different flavors of the same thing, people hate kilo for some reason and people hate corg for good reason. That leaves 3 maps that at least I personally would consider good, 2 of which are exclusive to diametrically different player counts. We could really use something more unique.

## Testing Photographs and Procedure
<!-- Include any screenshots/videos/debugging steps of the modified code functioning successfully, ideally including edge cases. -->
<details>
<summary>Screenshots&Videos</summary>

(Use the map diff bot in the Checks tab for the most up to date pics)

![image](https://github.com/user-attachments/assets/fa6a7ca9-d543-443e-b38d-83ca597a5f00)

![StrongDMM-2025-01-02 16 03 10](https://github.com/user-attachments/assets/686c6d6b-b18e-4c7d-a295-c539522864b3)

![image](https://github.com/user-attachments/assets/fd34ec07-07fc-4a2d-a017-c4dcd05299eb)

![image](https://github.com/user-attachments/assets/92b49a19-862c-46cb-b831-69ee8f5e82b7)

![image](https://github.com/user-attachments/assets/efa79683-00b6-465d-821a-ba9d898fa2b5)

![image](https://github.com/user-attachments/assets/5d830768-c4c6-45ba-b9d5-0b3cb49e9555)

![image](https://github.com/user-attachments/assets/564c4c39-1dd9-4c56-b6bb-b1bcc283b6c6)

![image](https://github.com/user-attachments/assets/3df1fd0a-6ef9-4a97-92e5-192c97beeb5c)

![image](https://github.com/user-attachments/assets/95320c7b-e11a-4fc4-977b-16ce346ed16a)

![image](https://github.com/user-attachments/assets/cd525b96-f948-4a94-a3cd-85a2a4340084)

![image](https://github.com/user-attachments/assets/096acb0d-e5bc-437c-b1a5-5db7b134fcd3)

![image](https://github.com/user-attachments/assets/a3c1f8bc-1ba5-4140-a72c-7d52151d9749)

</details>

## Changelog
:cl:
add: Added Card Station and all of it's shuttles
add: Added big potted plants
add: Added sub divided big engines for shuttles
add: Added a Cardinal bird mob
fix: fixed decals not working on circuit and grass floor tiles
fix: fixed fence gates not working
fix: fixed barricades' layer being below the layer of airlocks
tweak: adjusted some decals to be more consistent 
tweak: made a bunch of item types not get their random offset applied on map load if their position has been set berfore hand
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

ClownMoff commented on _maps/bunchedstation.json:
```diff
@@ -0,0 +1,12 @@
+{
+	"map_name": "Bunched Station",
```
Is this the final name of the station? Maybe you could take a name from here that hasn't been used.

https://en.wikipedia.org/wiki/NATO_phonetic_alphabet

BarteG44 replied:
see comment above

BarteG44 replied:
(forgor to push commit)

ClownMoff commented on code/datums/shuttles.dm:
```diff
@@ -379,6 +379,12 @@
+	suffix = "bunched"
+	name = "Bunched Station Emergency Shuttle"
+	credit_cost = 2000
+	description = "Bazinga."
```
Give it an actual description that makes sense IC

BarteG44 replied:
yep, placeholder. I'll have something here by the evening

ClownMoff commented on config/maps.txt:
```diff
@@ -56,6 +56,11 @@ map echostation
 endmap
 
+map bunchedstation
+	minplayers 20
```
```suggestion
	minplayers 40
```


Bump this to at least the same amount of players required for Fland. This is a huge station and it is unrealistic to think that this map will play well with 20 people. I don't think it will play well with 30 either but 20 is just straight wrong. If the map gets voted with 25 people, 5 leave, 20 come into the round, 2 people go mining and 2 exploring you end up with 16 people in a station that should be for +40 people **on station**

BarteG44 replied:
again, as bacon said, it's not just the footprint of the map that matters. We'll have to see how it actually plays in game. If it does feel too empty then i'm fine with increasing it

ClownMoff replied:
Yeah, both matter, that's why this map can feel amazing but if I'm alone wandering in hallways 6 tiles wide by myself and I see some assistant from time to time the map will fail as a lowpop map. Echo can get away with this because everything is cramped together, Delta puts 40 people minimun, same with Fland. Box can get away with being a big map for lowpop since the majority of the action happens on the same hallway (this map has those points of interest/actions either in smaller size or more separated between them so it wont get the same Box effect). 20 pop is a bad pop to put on a map this big and I think the grand majority of mappers would agree and suggest, at minimun, 25-30.

Tsar-Salat replied:
> Bump this to at least the same amount of players required for Fland.

This wasn't even tested at this point.

ClownMoff reviewed (dismissed):
My thoughts on some of the code

Tsar-Salat reviewed (dismissed):
Add navigation markers and I will put it into TM again

Tsar-Salat committed:
invalid backpack

BarteG44 committed:
makes nav markers beterer

BarteG44 committed:
small stuff

BarteG44 committed:
removes a stray SEC sign that no one ever mentioned

MarkusLarsson421 commented on _maps/_basemap.dm:
```diff
@@ -1,9 +1,10 @@
 // uncomment this for a map you need to use
 // #define FORCE_MAP "corgstation"
 // #define FORCE_MAP "boxstation"
+#define FORCE_MAP "cardstation"
```
This should be commented out prior to merge, right? Just a reminder.

BarteG44 replied:
Yes, it's forced for the duration of the TMs

BarteG44 committed:
review fixes

BarteG44 committed:
minor area change

BarteG44 committed:
apc whoopsie

BarteG44 committed:
cursed by the eternal icondiff

BarteG44 committed:
exploshutle de55plasmasheeting

BarteG44 committed:
come on lints

BarteG44 committed:
laeyr fix

BarteG44 committed:
more details

BarteG44 committed:
maybe now?

BarteG44 committed:
please?

BarteG44 committed:
damn you

BarteG44 committed:
this has got to be it

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
test

BarteG44 committed:
why are you doing this to me

BarteG44 committed:
why

BarteG44 committed:
hate you

Tsar-Salat committed:
fuck it

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into atom-storage

Tsar-Salat committed:
fix 2 things

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
merge conflict fix and a paper fix

BarteG44 committed:
i am going to scream

Tsar-Salat committed:
fix colorizer

BarteG44 committed:
security wallmounts

Tsar-Salat committed:
steal from modsuits branch to fix this bug... again

BarteG44 committed:
adds an actual welding mask to robotics instead of the prop (whoops)

Tsar-Salat committed:
readd ruko's trashbag nerf (sorta), readds kirbyplants storage

https://github.com/tgstation/tgstation/pull/73761

DrDuckedGoose committed:
Merge branch 'master' into xenoarch_redux

Tsar-Salat committed:
fix plant bags, and backpack weight limits

https://github.com/tgstation/tgstation/pull/68842

Tsar-Salat committed:
badstupid EVIL test!!!

Tsar-Salat committed:
Merge remote-tracking branch 'BeeStation/master' into coffeeshopdlc

# Conflicts:
#	_maps/map_files/CorgStation/CorgStation.dmm

BriggsIDP committed:
map fixes and delta

BriggsIDP committed:
lints v1

BriggsIDP committed:
lint v2

BriggsIDP committed:
v3 maybe this will fix it

BriggsIDP committed:
v4 dorms!

BriggsIDP committed:
One must imagine Sisyphus happy

DrDuckedGoose committed:
Update sentient.dm

DrDuckedGoose committed:
Update haunted.dm

DrDuckedGoose committed:
Update animalize.dm

DrDuckedGoose committed:
Update xenoarchaeology_machines.dm

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
resolves conflict

BarteG44 committed:
it was meant to be a soft removal

BarteG44 committed:
explo shuttle fix

DrDuckedGoose committed:
Merge branch 'master' into xenoarch_redux

Tsar-Salat committed:
Merge remote-tracking branch 'origin/master' into atom-storage

# Conflicts:
#	code/modules/food_and_drinks/kitchen_machinery/griddle.dm

Tsar-Salat committed:
readd cig interaction

https://github.com/tgstation/tgstation/pull/72104

Tsar-Salat committed:
woops

Tsar-Salat committed:
not needed

Tsar-Salat committed:
fix hop

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into atom-storage

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
Merge branch 'pr/12358' into bunched

DrDuckedGoose committed:
Merge branch 'master' into xenoarch_redux

DrDuckedGoose committed:
Fix corgi qdel - 6 3 25

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into atom-storage

# Conflicts:
#	code/modules/antagonists/abductor/equipment/abduction_gear.dm

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into coffeeshopdlc

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into atom-storage

BarteG44 committed:
Adds the Caffe and reorganizes medbay

BarteG44 committed:
small fixes

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into coffeeshopdlc

Tsar-Salat committed:
addressing

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into coffeeshopdlc

DrDuckedGoose committed:
Merge branch 'master' into xenoarch_redux

DrDuckedGoose committed:
Update animalize.dm

DrDuckedGoose committed:
Ya like dags? - 9 3 25

BarteG44 committed:
Ports <Dmicon> and adds wall closets

BarteG44 committed:
messed up while copy pasting

BarteG44 committed:
Technically works

BarteG44 committed:
idk man i don't get this

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
Merge branch 'port-of-a-port-of-a-port-of-dmicon' into bunched

BarteG44 committed:
merge conflict

DrDuckedGoose committed:
Merge branch 'master' into xenoarch_redux

DrDuckedGoose committed:
bump

Tsar-Salat committed:
Merge branch 'master' into atom-storage

BarteG44 committed:
i guess

Tsar-Salat committed:
Merge remote-tracking branch 'upstream/master' into atom-storage

Tsar-Salat committed:
.

BarteG44 committed:
lints?

BarteG44 committed:
lints?

also made the edge cases work with cringe snowflake code and icon2base64. Hopefully someone lets me know of a better solution

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
prettier also screwdrivers work now but are jank

BarteG44 committed:
wider conditions

BarteG44 committed:
Merge branch 'port-of-a-port-of-a-port-of-dmicon' into bunched

BarteG44 committed:
no more icon2base64

thanks absolucy

BarteG44 committed:
Merge branch 'port-of-a-port-of-a-port-of-dmicon' into bunched

BarteG44 committed:
wall closet integration

BarteG44 committed:
changes to some wall closets

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
details

BarteG44 committed:
more details

BarteG44 committed:
Merge branch 'pr/12358' into bunched

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into pr/12358

BarteG44 committed:
resolves conflicts

BarteG44 committed:
Merge branch 'pr/12358' into bunched

BarteG44 committed:
one (1) light bulb

BarteG44 committed:
one (1) cup

BarteG44 committed:
updates the icons AFTER initialize

BarteG44 committed:
Merge branch 'port-of-a-port-of-a-port-of-dmicon' into bunched

BarteG44 committed:
details

BarteG44 committed:
removes the random offset from paper on initialize.

 If a paper's meant to be offset on mapload, then that's up to the mapper to do. This just breaks the small decorations

BarteG44 committed:
adds a command closet

BarteG44 committed:
Merge branch 'port-of-a-port-of-a-port-of-dmicon' into bunched

BarteG44 committed:
adds a few command wall closets

BarteG44 committed:
basically a typo

BarteG44 committed:
Merge branch 'port-of-a-port-of-a-port-of-dmicon' into bunched

BarteG44 committed:
reverts the paper change

there is a separate PR for it now

BarteG44 committed:
whitespace

BarteG44 committed:
you can now take them apart. revolutionary

BarteG44 committed:
Merge branch 'port-of-a-port-of-a-port-of-dmicon' into bunched

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
resolves merge conflict

BarteG44 committed:
Revert "resolves merge conflict"

This reverts commit 6b6d38b563e069709bce80b839958b533762612d.

BarteG44 committed:
Revert "Merge remote-tracking branch 'upstream/master' into bunched"

This reverts commit aea8726f6dab039506da60e5cc4f6e975d61e827, reversing
changes made to 746efac1b959e5740bc02a5b328a56c816ddb712.

BarteG44 committed:
what is going on

BarteG44 committed:
Revert "Merge remote-tracking branch 'upstream/master' into bunched"

This reverts commit 746efac1b959e5740bc02a5b328a56c816ddb712, reversing
changes made to 8c13b4947967345eb665f82ed7157d9af9bd8a83.

BarteG44 committed:
Merge branch 'pr/11894' into bunched

BarteG44 committed:
TECHNICALLY works

I hope this will all resolve itself once wall closets and coffee shop PRs get merged

BarteG44 committed:
lint?

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
adds more external airlocks

BarteG44 committed:
one (1) camera

BarteG44 committed:
adds some comments

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
adds the new beakers to chemistry and around the station

BarteG44 committed:
oversight

BarteG44 committed:
details

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
Adds sounds to closet interactions

BarteG44 committed:
Merge branch 'port-of-a-port-of-a-port-of-dmicon' into bunched

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
MapDiffBot, please

DrDuckedGoose committed:
Merge branch 'master' into xenoarch_redux

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

DrDuckedGoose committed:
x gon give it to ya - 27 3 25

BarteG44 committed:
hopefully discards the changes to all other maps than card

BarteG44 committed:
what is even failing the lints here

BarteG44 committed:
reverts stuff

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
details

BarteG44 committed:
forgot an isliving check for sound

BarteG44 committed:
Merge branch 'port-of-a-port-of-a-port-of-dmicon' into bunched

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
light

BarteG44 committed:
what

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
help

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
adds closets to all maps

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
meata

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
small changes

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

DrDuckedGoose committed:
Fix runtimes & Bug/s - 9 4 25

DrDuckedGoose committed:
Merge branch 'master' into xenoarch_redux

BarteG44 committed:
fixes icons being blurry on 516

DrDuckedGoose committed:
bump

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
Allows borgs to look into the lockers, but not to interact with them.

Also stealth fixes the teleporter UI not auto updating, causing the calibration status not being obvious

BarteG44 committed:
Merge branch 'port-of-a-port-of-a-port-of-dmicon' into bunched

BarteG44 committed:
fixes some small issues

BarteG44 committed:
fixes some mapping issues

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
undo rad stuff

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
fixes cows

BarteG44 committed:
replaces funky wires with boring wires

[*]

BarteG44 committed:
shuttles too

BarteG44 committed:
details

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into port-of-a-port-of-a-port-of-dmicon

BarteG44 committed:
removes changes to maps

BarteG44 committed:
fixes procs and for loops

BarteG44 committed:
removes the semicolons

why were they even there

BarteG44 committed:
undoes changes done by the coffee PR

BarteG44 committed:
removes the rest of coffee stuff

BarteG44 committed:
Merge branch 'port-of-a-port-of-a-port-of-dmicon' into bunched

BarteG44 committed:
last remaining traces of the coffee shop

It will be back once that PR gets merged

BarteG44 committed:
makes barricades show up above doors because it makes more sense

BarteG44 committed:
plant name

this is the stuff i actually spend time on

BarteG44 committed:
puts the lockers into the shrink-inator

BarteG44 committed:
this offset looks better

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
Merge branch 'port-of-a-port-of-a-port-of-dmicon' into bunched

BarteG44 committed:
adds more wall lockers and fixes that one broken maint room

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
linter

BarteG44 committed:
fixes closet proximity checking and adds an open state sprite

BarteG44 committed:
adjusts the sprites

BarteG44 committed:
fixed multiple people logic

BarteG44 committed:
Merge branch 'port-of-a-port-of-a-port-of-dmicon' into bunched

BarteG44 committed:
a few details

BarteG44 committed:
engi changes

BarteG44 committed:
removes the FORCE_MAP

the day nears

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
why art thou failing checks

BarteG44 committed:
details

BarteG44 committed:
addresses review

BarteG44 committed:
lint

BarteG44 committed:
singular fire alarm

DrDuckedGoose committed:
Merge branch 'master' into xenoarch_redux

DrDuckedGoose committed:
Update RadStation.dmm

BarteG44 committed:
forgor to fix the decals

BarteG44 committed:
review part 2

DrDuckedGoose committed:
Merge branch 'master' into xenoarch_redux

Tsar-Salat reviewed (approved):
Code is fine. Floria has covered map review for me.

spockye reviewed (changes requested):
![image](https://github.com/user-attachments/assets/2ed76dd4-6904-4a1e-a04f-cd1e46bb7346)

seems unintentional?

![image](https://github.com/user-attachments/assets/3fe19c9a-d1d4-4844-bed4-31217dd0f8d7)

in Genetics

-----
gotta say man, this is some amazing work. love how it looks, and the attention to detail is on point

would love to see it in action 
keep it up!

BarteG44 committed:
fixes pipe and area

BarteG44 committed:
forgor

DrDuckedGoose committed:
Fixes - 26 4 25

DrDuckedGoose committed:
Disable animal trait - 26 4 25

BarteG44 committed:
sovl

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
guitar detail plus mapped holochip conditions

BarteG44 committed:
pixelshifts items on the tables in the pool room

DrDuckedGoose committed:
Merge branch 'master' into xenoarch_redux

DrDuckedGoose committed:
map shit bot

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
small stuff

BarteG44 committed:
beret i guess

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
Merge branch 'pr/10906' into bunched

BarteG44 committed:
updates artsci to xenoarch 2

PestoVerde322 reviewed (dismissed):
First off, the name of the map file name should be updated to CardinalStation.dmm, to reflect the actual name, now onto various things that are still wrong:

![image](https://github.com/user-attachments/assets/7e330e7d-f779-49cd-8ed0-b568c6e092b8)
Fence in wall, looks off.

![image](https://github.com/user-attachments/assets/0571b99c-1b98-431e-8a5b-3890803473ca)
two indestructible walls?

![image](https://github.com/user-attachments/assets/6390a2e4-37c4-49ca-92a2-0f958aa45e3a)
Three different APCs for otherwise three unassuming areas, now I might be fuzzy on this one, but i recall that there's a way to tie the zone to an APC, ask on discord for further infos.

BarteG44 committed:
adresses change request

BarteG44 committed:
whoops

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
Update _basemap.dm

BarteG44 committed:
Update _basemap.dm

BarteG44 committed:
details

BarteG44 committed:
Update emergency_card.dmm

BarteG44 committed:
Merge remote-tracking branch 'upstream/master' into bunched

BarteG44 committed:
adds the cardinal bird

BarteG44 committed:
details

BarteG44 committed:
whole lotta pixel shifting
